newrelic/core/agent_control_health.py

```diff
@@ -130,13 +125,23 @@ def __init__(self):
 
     @property
     def health_check_enabled(self):
-        fleet_id_present = os.environ.get("NEW_RELIC_AGENT_CONTROL_FLEET_ID", None)
-        if not fleet_id_present:
+        # Default to False - this must be explicitly set to True by the sidecar/ operator to enable health check
+        agent_control_enabled = os.environ.get("NEW_RELIC_AGENT_CONTROL_ENABLED", False)
+        if not agent_control_enabled:
             return False
 
-        health_file_location = os.environ.get("NEW_RELIC_AGENT_CONTROL_HEALTH_DELIVERY_LOCATION", None)
+        return is_valid_file_delivery_location(self.health_delivery_location)
```

Comment:
Can we use `_environ_as_bool()` for this? This currently just checks if it's an empty string or not. It would be nice to be able to explicitly disable this by setting it to "false" or "0" etc. If we do that currently we're likely to get confused why it's still enabled.

```python
        from newrelic.core.config import _environ_as_bool

        agent_control_enabled = _environ_as_bool("NEW_RELIC_AGENT_CONTROL_ENABLED", False)
        if not agent_control_enabled:
            return False

        return is_valid_file_delivery_location(self.health_delivery_location)
```